docs/user-guide/arrays.rst

```diff
@@ -0,0 +1,487 @@
+
+The contents of the array can be retrieved by slicing, which will load the
+requested region into memory as a NumPy array, e.g.:
+
+.. ipython:: python
+
+   z[0, 0]
+   z[-1, -1]
+   z[0, :]
+   z[:, 0]
+   z[:]
+
+.. _tutorial_persist:
+
+Persistent arrays
+-----------------
+
+In the examples above, compressed data for each chunk of the array was stored in
+main memory. Zarr arrays can also be stored on a file system, enabling
+persistence of data between sessions. For example:
+
+.. ipython:: python
+
+   z1 = zarr.open('data/example.zarr', mode='w', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
+
+The array above will store its configuration metadata and all compressed chunk
+data in a directory called 'data/example.zarr' relative to the current working
+directory. The :func:`zarr.api.synchronous.open` function provides a convenient way
+to create a new persistent array or continue working with an existing
+array. Note that although the function is called "open", there is no need to
+close an array: data are automatically flushed to disk, and files are
+automatically closed whenever an array is modified.
+
+Persistent arrays support the same interface for reading and writing data,
+e.g.:
+
+.. ipython:: python
+
+   z1[:] = 42
+   z1[0, :] = np.arange(10000)
+   z1[:, 0] = np.arange(10000)
+
+Check that the data have been written and can be read again:
+
+.. ipython:: python
+
+   z2 = zarr.open('data/example.zarr', mode='r')
+   np.all(z1[:] == z2[:])
+
+If you are just looking for a fast and convenient way to save NumPy arrays to
+disk then load back into memory later, the functions
+:func:`zarr.convenience.save` and :func:`zarr.convenience.load` may be
+useful. E.g.:
+
+.. ipython::
+   :suppress:
+
+   In [144]: rm -r data/example.zarr
+
+.. ipython:: python
+
+   a = np.arange(10)
+   zarr.save('data/example.zarr', a)
+   zarr.load('data/example.zarr')
+
+Please note that there are a number of other options for persistent array
+storage, see the section on :ref:`tutorial_storage` below.
+
+.. _tutorial_resize:
+
+Resizing and appending
+----------------------
+
+A Zarr array can be resized, which means that any of its dimensions can be
+increased or decreased in length. For example:
+
+.. ipython:: python
+
+   z = zarr.zeros(shape=(10000, 10000), chunks=(1000, 1000))
+   z[:] = 42
+   z.resize((20000, 10000))
+   z.shape
+
+Note that when an array is resized, the underlying data are not rearranged in
+any way. If one or more dimensions are shrunk, any chunks falling outside the
+new array shape will be deleted from the underlying store.
+
+For convenience, Zarr arrays also provide an ``append()`` method, which can be
+used to append data to any axis. E.g.:
+
+.. ipython:: python
+
+   a = np.arange(10000000, dtype='i4').reshape(10000, 1000)
+   z = zarr.array(a, chunks=(1000, 100))
+   z.shape
+   z.append(a)
+   z.append(np.vstack([a, a]), axis=1)
+   z.shape
+
+.. _tutorial_compress:
+
+Compressors
```

Comment:
I think we should add a section for sharding.

Reply:
top-level note on the differences between V2 and V3 would be good. (I'm sure this is on your todo list, just a reminder)